QUARKUS-2391.md

```diff
@@ -0,0 +1,38 @@
+# QUARKUS-2391 Promote the OpenTelemetry extension from TP to full support
+JIRA:https://issues.redhat.com/browse/QUARKUS-2391
+## Scope of the testing
+Quarkus already supported Opentelemetry as a tech preview, but now we move towards full support, then we will extend our test coverage.
+### Existing tests
+In the current quarkus-test-suite exist two modules dedicated to opentelemetry in the monitoring folder, they are:
+
+- opentelemetry [monitoring-opentelemetry]
+- opentelemetry-reactive [monitoring-opentelemetry-reactive]
+
+These tests utilize trace export to Jaeger (all-in-one pod) components and context propagation through a Ping-Pong application with 3 pods (ping service,pong service, jaeger-all-in-one).
+The traces are directly sent to the Jaeger collector, and they also test Quarkus gRPC communication between services and SSE (Server-Sent Events).
+
+On the other hand, also other modules use ```quarkus-opentelemetry``` dependency:
+- javaee-like-getting-started
+- build-time-analytics
+- http/graphql-telemetry
+- http/vertx-web-client
+- sql-db/narayana-transactions (where is tested JDBC Opentelemetry instrumentation)
+
+#### Test scenarios
+- Disable Opentelemetry extensions to ensure the application behaves as expected without tracing enabled.
+- Testing scenario with additional configuration options such as Id Generator, Propagators, Resource, Sampler configuration.
+- Test with other exporters such a Zipkin.
+- Integration tests to ensure proper tracing is enabled and functioning correctly in other core Quarkus extensions such as ```quarkus-smallrye-reactive-messaging-amqp```.
+
+### Impact on testsuites and testing automation:
+New tests will be added inside monitoring-opentelemetry modules and messaging-amqp-reactive.
+And ensure that test cases are running in both JVM and OpenShift environments.
```

Comment:
We usually have matrix of 4 environments: JVM/Native and Baremetal/Openshift. Which are we going to use there?

Reply:
I wrote down JVM (I think by default all the tests are running in bare metal/JVM mode)  and Openshift (keeping the monitoring Opentelemetry testing strategy). Not sure, if in this case, we can provide some value also to include Native execution.

Reply:
When, please, write this more clearly, "running on JVM in both baremetal and OpenShift environmeents"

Reply:
I apologize for any confusion, I've corrected the sentence for clarification, thanks @fedinskiy .